### ISSUE TITLE
Frequent I2C transmit errors when I2S is active, chip ESP32 S3 (IDFGH-9954)

### ISSUE THREAD

**Issue by hyunel** [espressif/esp-idf#11247]
### Answers checklist.

- [X] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [X] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### IDF version.

v5.1-dev-4726-gdf9310ada2

### Operating System used.

Linux

### How did you build your project?

Command line with idf.py

### If you are using Windows, please specify command line type.

None

### Development Kit.

Custom Board

### Power Supply used.

USB

### What is the expected behavior?

I2C transmits data without any errors

### What is the actual behavior?

When I2S channel is enabled, I2C transmission has frequent errors (and sometimes reads the wrong value)

### Steps to reproduce.

Here is my test code (If everything is correct, the value read by I2C will be 191):

```c
#include <stdio.h>
#include <inttypes.h>
#include "sdkconfig.h"
#include "esp_chip_info.h"
#include "esp_flash.h"
#include "driver/i2s_std.h"
#include "driver/i2c.h"
#include "esp_log.h"

const char *TAG = "main";

void app_main(void)
{
    ESP_LOGI(TAG, "Hello world!");
    
    /* Print chip information */
    esp_chip_info_t chip_info;
    uint32_t flash_size;
    esp_chip_info(&chip_info);
    ESP_LOGI(TAG, "This is %s chip with %d CPU core(s), WiFi%s%s, ",
           CONFIG_IDF_TARGET,
           chip_info.cores,
           (chip_info.features & CHIP_FEATURE_BT) ? "/BT" : "",
           (chip_info.features & CHIP_FEATURE_BLE) ? "/BLE" : "");

    unsigned major_rev = chip_info.revision / 100;
    unsigned minor_rev = chip_info.revision % 100;
    ESP_LOGI(TAG, "silicon revision v%d.%d, ", major_rev, minor_rev);
    if(esp_flash_get_size(NULL, &flash_size) != ESP_OK) {
        ESP_LOGI(TAG, "Get flash size failed");
        return;
    }
    
    ESP_LOGI(TAG, "Minimum free heap size: %" PRIu32 " bytes", esp_get_minimum_free_heap_size());
    
    i2s_std_config_t std_cfg = {
        .clk_cfg = {
            .sample_rate_hz = 44100,
            .clk_src = I2S_CLK_SRC_DEFAULT,
            .mclk_multiple = 384,
        },
        .slot_cfg = I2S_STD_PHILIPS_SLOT_DEFAULT_CONFIG(24, I2S_SLOT_MODE_STEREO),
        .gpio_cfg = {
            .mclk = 7,
            .bclk = 6,
            .ws = 2,
            .dout = 5,
            .din = -1
        },
    };
    i2s_chan_config_t chan_cfg = I2S_CHANNEL_DEFAULT_CONFIG(I2S_NUM_0, I2S_ROLE_MASTER);
    i2s_chan_handle_t tx_handle = NULL;
    ESP_ERROR_CHECK(i2s_new_channel(&chan_cfg, &tx_handle, NULL));
    ESP_ERROR_CHECK(i2s_channel_init_std_mode(tx_handle, &std_cfg));
    ESP_ERROR_CHECK(i2s_channel_enable(tx_handle));

    const i2c_config_t i2c_cfg = {
        .sda_io_num = 8,
        .scl_io_num = 9,
        .mode = I2C_MODE_MASTER,
        .sda_pullup_en = GPIO_PULLUP_ENABLE,
        .scl_pullup_en = GPIO_PULLUP_ENABLE,
        .master.clk_speed = 100000,
    };

    ESP_ERROR_CHECK(i2c_param_config(I2C_NUM_0, &i2c_cfg));
    ESP_ERROR_CHECK(i2c_driver_install(I2C_NUM_0, i2c_cfg.mode, 0, 0, 0));

    const uint8_t dev_addr = 0x08;
    while (1)
    {
        esp_err_t err = ESP_OK;
        i2c_cmd_handle_t cmd = i2c_cmd_link_create();

        err |= i2c_master_start(cmd);
        err |= i2c_master_write_byte(cmd, (dev_addr << 1) | I2C_MASTER_WRITE, 1);
        err |= i2c_master_write_byte(cmd, 0x14, 1);

        err |= i2c_master_start(cmd);
        err |= i2c_master_write_byte(cmd, (dev_addr << 1) | I2C_MASTER_READ, 1);

        uint8_t data;
        err |= i2c_master_read(cmd, &data, 1, I2C_MASTER_LAST_NACK);
        err |= i2c_master_stop(cmd);
        err |= i2c_master_cmd_begin(I2C_NUM_0, cmd, 200 / portTICK_PERIOD_MS);
        i2c_cmd_link_delete(cmd);

        if (err != ESP_OK)
        {
            ESP_LOGE(TAG, "E: %d", err);
        } else if (data != 191) {
            ESP_LOGW(TAG, "V: %d", data);
        } else {
            ESP_LOGI(TAG, "P");
        }
    }
}
```

### Debug Logs.

```plain
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x1 (POWERON),boot:0x9 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3818,len:0x16f4
load:0x403c9700,len:0x4
load:0x403c9704,len:0xbfc
load:0x403cc700,len:0x2ee0
entry 0x403c9904
I (27) boot: ESP-IDF v5.1-dev-4726-gdf9310ada2 2nd stage bootloader
I (27) boot: compile time Apr 21 2023 18:16:03
I (28) boot: Multicore bootloader
I (32) boot: chip revision: v0.1
I (35) boot.esp32s3: Boot SPI Speed : 80MHz
I (40) boot.esp32s3: SPI Mode       : DIO
I (45) boot.esp32s3: SPI Flash Size : 4MB
I (50) boot: Enabling RNG early entropy source...
I (55) boot: Partition Table:
I (59) boot: ## Label            Usage          Type ST Offset   Length
I (66) boot:  0 nvs              WiFi data        01 02 00009000 00006000
I (73) boot:  1 phy_init         RF data          01 01 0000f000 00001000
I (81) boot:  2 factory          factory app      00 00 00010000 00100000
I (88) boot: End of partition table
I (93) esp_image: segment 0: paddr=00010020 vaddr=3c020020 size=0c778h ( 51064) map
I (110) esp_image: segment 1: paddr=0001c7a0 vaddr=3fc93200 size=028dch ( 10460) load
I (113) esp_image: segment 2: paddr=0001f084 vaddr=40374000 size=00f94h (  3988) load
I (119) esp_image: segment 3: paddr=00020020 vaddr=42000020 size=1ddc0h (122304) map
I (148) esp_image: segment 4: paddr=0003dde8 vaddr=40374f94 size=0e1a4h ( 57764) load
I (168) boot: Loaded app from partition at offset 0x10000
I (168) boot: Disabling RNG early entropy source...
I (179) cpu_start: Multicore app
I (179) esp_psram: Found 2MB PSRAM device
I (180) esp_psram: Speed: 80MHz
I (180) cpu_start: Pro cpu up.
I (184) cpu_start: Starting app cpu, entry point is 0x403753cc
0x403753cc: call_start_cpu1 at /mnt/Others/Repository/ToolChain/esp-idf/components/esp_system/port/cpu_start.c:159

I (0) cpu_start: App cpu up.
I (420) esp_psram: SPI SRAM memory test OK
I (429) cpu_start: Pro cpu start user code
I (435) cpu_start: cpu freq: 240000000 Hz
I (435) cpu_start: Application information:
I (438) cpu_start: Project name:     hello_world
I (443) cpu_start: App version:      1
I (447) cpu_start: Compile time:     Apr 21 2023 18:15:32
I (453) cpu_start: ELF file SHA256:  5fb8ab9a44cba6d4...
I (459) cpu_start: ESP-IDF:          v5.1-dev-4726-gdf9310ada2
I (466) cpu_start: Min chip rev:     v0.0
I (471) cpu_start: Max chip rev:     v0.99 
I (475) cpu_start: Chip rev:         v0.1
I (480) heap_init: Initializing. RAM available for dynamic allocation:
I (487) heap_init: At 3FC96378 len 00053398 (332 KiB): DRAM
I (494) heap_init: At 3FCE9710 len 00005724 (21 KiB): STACK/DRAM
I (500) heap_init: At 3FCF0000 len 00008000 (32 KiB): DRAM
I (506) heap_init: At 600FE010 len 00001FF0 (7 KiB): RTCRAM
I (513) esp_psram: Adding pool of 2048K of PSRAM memory to heap allocator
I (521) spi_flash: detected chip: generic
I (525) spi_flash: flash io: dio
I (529) sleep: Configure to isolate all GPIO pins in sleep state
I (536) sleep: Enable automatic switching of GPIO sleep configuration
I (543) app_start: Starting scheduler on CPU0
I (548) app_start: Starting scheduler on CPU1
I (548) main_task: Started on CPU0
I (558) esp_psram: Reserving pool of 32K of internal memory for DMA/internal allocations
I (568) main_task: Calling app_main()
I (568) main: Hello world!
I (578) main: This is esp32s3 chip with 2 CPU core(s), WiFi/BLE, 
I (578) main: silicon revision v0.1, 
I (588) main: Minimum free heap size: 2444456 bytes
E (598) main: E: -1
I (598) main: P
I (598) main: P
I (598) main: P
E (598) main: E: -1
I (608) main: P
E (608) main: E: -1
I (608) main: P
I (608) main: P
E (618) main: E: -1
E (618) main: E: -1
I (618) main: P
I (618) main: P
I (628) main: P
I (628) main: P
E (628) main: E: -1
I (628) main: P
I (638) main: P
E (638) main: E: -1
E (638) main: E: -1
E (638) main: E: -1
I (648) main: P
I (648) main: P
I (648) main: P
I (648) main: P
I (658) main: P
E (658) main: E: -1
I (658) main: P
I (658) main: P
I (668) main: P
I (668) main: P
E (668) main: E: -1
E (668) main: E: -1
E (678) main: E: -1
I (678) main: P
I (678) main: P
I (678) main: P
I (688) main: P
I (688) main: P
I (688) main: P
I (688) main: P
I (698) main: P
I (698) main: P
E (698) main: E: -1
I (698) main: P
I (708) main: P
I (708) main: P
I (708) main: P
I (708) main: P
I (718) main: P
I (718) main: P
I (718) main: P
I (718) main: P
E (728) main: E: -1
I (728) main: P
I (728) main: P
E (728) main: E: -1
I (738) main: P
I (738) main: P
I (738) main: P
I (738) main: P
E (748) main: E: -1
I (748) main: P
I (748) main: P
I (748) main: P
I (758) main: P
I (758) main: P
I (758) main: P
I (758) main: P
I (768) main: P
I (768) main: P
I (768) main: P
I (768) main: P
I (778) main: P
I (778) main: P
W (778) main: V: 127
I (778) main: P
I (788) main: P
I (788) main: P
I (788) main: P
I (788) main: P
I (798) main: P
I (798) main: P
I (798) main: P
I (798) main: P
I (808) main: P
I (808) main: P
I (808) main: P
I (808) main: P
I (818) main: P
I (818) main: P
I (818) main: P
I (818) main: P
I (828) main: P
I (828) main: P
I (828) main: P
I (828) main: P
I (828) main: P
E (838) main: E: -1
I (838) main: P
E (838) main: E: -1
I (848) main: P
I (848) main: P
I (848) main: P
I (848) main: P
E (848) main: E: -1
E (858) main: E: -1
I (858) main: P
E (858) main: E: -1
I (868) main: P
I (868) main: P
I (868) main: P
I (868) main: P
I (878) main: P
I (878) main: P
I (878) main: P
I (878) main: P
I (888) main: P
I (888) main: P
I (888) main: P
I (888) main: P
E (888) main: E: -1
I (898) main: P
I (898) main: P
I (898) main: P
I (908) main: P
I (908) main: P
I (908) main: P
I (908) main: P
I (908) main: P
I (918) main: P
E (918) main: E: -1
I (918) main: P
I (918) main: P
I (928) main: P
I (928) main: P
I (928) main: P
I (928) main: P
I (938) main: P
I (938) main: P
I (938) main: P
I (938) main: P
I (948) main: P
I (948) main: P
I (948) main: P
W (948) main: V: 127
I (958) main: P
I (958) main: P
I (958) main: P
I (958) main: P
I (968) main: P
I (968) main: P
I (968) main: P
I (968) main: P
I (978) main: P
I (978) main: P
I (978) main: P
I (978) main: P
I (988) main: P
I (988) main: P
I (988) main: P
I (988) main: P
I (998) main: P
W (998) main: V: 127
I (998) main: P
I (998) main: P
I (1008) main: P
I (1008) main: P
I (1008) main: P
I (1008) main: P
I (1018) main: P
I (1018) main: P
I (1018) main: P
I (1018) main: P
I (1028) main: P
I (1028) main: P
I (1028) main: P
I (1028) main: P
I (1038) main: P
E (1038) main: E: -1
I (1038) main: P
I (1038) main: P
I (1048) main: P
I (1048) main: P
I (1048) main: P
I (1048) main: P
I (1058) main: P
I (1058) main: P
I (1058) main: P
I (1058) main: P
I (1068) main: P
I (1068) main: P
I (1068) main: P
W (1068) main: V: 127
I (1078) main: P
I (1078) main: P
I (1078) main: P
I (1078) main: P
E (1088) main: E: -1
I (1088) main: P
I (1088) main: P
I (1098) main: P
I (1098) main: P
I (1098) main: P
I (1098) main: P
I (1108) main: P
I (1108) main: P
E (1108) main: E: -1
I (1108) main: P
I (1118) main: P
I (1118) main: P
I (1118) main: P
I (1118) main: P
I (1128) main: P
E (1128) main: E: -1
I (1128) main: P
I (1128) main: P
I (1138) main: P
I (1138) main: P
I (1138) main: P
I (1138) main: P
I (1148) main: P
I (1148) main: P
I (1148) main: P
I (1148) main: P
I (1158) main: P
I (1158) main: P
I (1158) main: P
I (1158) main: P
I (1168) main: P
I (1168) main: P
E (1168) main: E: -1
I (1168) main: P
E (1178) main: E: -1
I (1178) main: P
I (1178) main: P
I (1178) main: P
```

### More Information.

 - When `i2s_channel_enable(tx_handle)` is commented out, then no error occurs.
 - Tried changing the I2S clock source, but it didn't work.
 - I found that turning down `mclk_multiple` (e.g. to 256) resulted in a significant reduction in error frequency.

**Comment by toanadas1** [comment]
Hi @hyunel 
I'm encountering the same issue with I2S and I2C, could you please point out at which commit this bug has been fixed?

**Comment by hyunel** [comment]
> Hi @hyunel I'm encountering the same issue with I2S and I2C, could you please point out at which commit this bug has been fixed? ESP-IDF: v5.1.1

This was supposed to be my pcb design problem, which I finally solved by adding I2C pull-up resistors.

**Comment by toanadas1** [comment]
@hyunel Thanks for the information
